src/wrapperInfo.ts

```diff
@@ -59,7 +59,7 @@ class WrapperInfo implements IWrapperInfo {
 
     core.debug(`distributionUrl: ${distributionUrl}`);
 
-    const parsed = /^distributionUrl=.*\/gradle-([^-]+)-([^.]+)\.zip$/.exec(
+    const parsed = /^distributionUrl=.*\/gradle-(.+)-([^.-]+)\.zip$/.exec(
```

Comment:
This failed for `gradle-7.0-rc-1-all.zip`.

Reply:
Gradle's files naming seem to be quite consistent, what about using a non-capturing group to ignore the `rc-X` part?

```js
/^distributionUrl=.*\/gradle-([^-]+)-(?:rc-[0-9]+)-([^.]+)\.zip$/.exec(...)
```